pkg/maps/lbmap/maglev.go

```diff
@@ -14,13 +14,153 @@
 
 package lbmap
 
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/byteorder"
+	"github.com/cilium/cilium/pkg/option"
+)
+
 const (
 	// Both inner maps are not being pinned into BPF fs.
-	MaglevInner4MapName = "cilium_lb4_maglev_meta"
-	MaglevInner6MapName = "cilium_lb6_maglev_meta"
+	MaglevInner4MapName = "cilium_lb4_maglev_inner"
+	MaglevInner6MapName = "cilium_lb6_maglev_inner"
 
 	// Both outer maps are pinned though given we need to attach
 	// inner maps into them.
 	MaglevOuter4MapName = "cilium_lb4_maglev"
 	MaglevOuter6MapName = "cilium_lb6_maglev"
 )
+
+var MaglevOuter4Map *bpf.Map
+var MaglevOuter6Map *bpf.Map
+
+func InitMaglevMaps(ipv4 bool, ipv6 bool) error {
+	dummyInnerMap := newInnerMaglevMap("cilium_lb_maglev_dummy")
+	if _, err := dummyInnerMap.OpenOrCreateUnpinned(); err != nil {
+		return err
+	}
+	defer dummyInnerMap.Close()
+
+	if ipv4 {
+		MaglevOuter4Map = newOuterMaglevMap(MaglevOuter4MapName, dummyInnerMap)
+		if _, err := MaglevOuter4Map.OpenOrCreate(); err != nil {
+			return err
+		}
+	}
+	if ipv6 {
+		MaglevOuter6Map = newOuterMaglevMap(MaglevOuter6MapName, dummyInnerMap)
+		if _, err := MaglevOuter6Map.OpenOrCreate(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func newInnerMaglevMap(name string) *bpf.Map {
+	return bpf.NewMap(
+		name,
+		bpf.MapTypeArray,
+		&MaglevInnerKey{}, int(unsafe.Sizeof(MaglevInnerKey{})),
+		&MaglevInnerVal{}, 2*option.Config.MaglevTableSize,
```

Comment:
Leave a comment here why it's using `2* option.Config.MaglevTableSize`

Reply:
Makes sense, it's for the __u16 size. We'll do.